public/app/plugins/datasource/grafana-pyroscope-datasource/VariableQueryEditor.tsx

```diff
@@ -30,23 +30,22 @@ export function VariableQueryEditor(props: QueryEditorProps<PyroscopeDataSource,
             onChange={(value) => {
               if (value.value! === 'profileType') {
                 props.onChange({
-                  ...props.query,
                   type: value.value!,
+                  refId: props.query.refId,
                 });
               }
               if (value.value! === 'label') {
                 props.onChange({
-                  ...props.query,
                   type: value.value!,
-                  profileTypeId: '',
+                  refId: props.query.refId,
+                  profileTypeId: props.query.type !== 'profileType' ? props.query.profileTypeId : '',
                 });
               }
               if (value.value! === 'labelValue') {
                 props.onChange({
-                  ...props.query,
                   type: value.value!,
-                  profileTypeId: '',
-                  labelName: '',
+                  refId: props.query.refId,
+                  profileTypeId: props.query.type !== 'profileType' ? props.query.profileTypeId : '',
                 });
```

Comment:
Is this the same block above—the one for the `value.value! === 'label'` case? What about merging the two then?
```
if (value.value! === 'label' || value.value! === 'labelValue') {
  props.onChange({
    type: value.value!,
    refId: props.query.refId,
    // Make sure we keep already selected values if they make sense for the variable type
    profileTypeId: props.query.type !== 'profileType' ? props.query.profileTypeId : '',
  });
}
```